### PR TITLE
Feedback improvements with new interface and new methods for ease of use

### PR DESCRIPTION
**Pull request by ijunaid**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## x.x.x
+* Added `CountlyFeedbacks` interface to manage feedback widgets, including NPS, Surveys, and Ratings.
+    * `presentNPS`, `presentSurvey`, and `presentRating` methods now allow displaying specific widgets by name, ID, or tag.
```
Comment (arifBurakDemiray) [inline CHANGELOG.md:3]: it is great but mb we could add function signatures with class only not with parameter names like
presentNPS(String)

**Review by arifBurakDemiray — COMMENTED** [review]
I will give my attitudes and good luck. You rock 🪨

File: CHANGELOG.md (hunk 1 above)
Comment (AliRKat) [reply CHANGELOG.md:3]: i understand why @arifBurakDemiray is suggesting that but current version also feels totally fine to me

File: CountlyFeedbackWidget.h [hunk 2]
```diff
@@ -41,6 +42,12 @@ extern NSString* const kCountlyReservedEventRating;
  */
 - (void)presentWithAppearBlock:(void(^ __nullable)(void))appearBlock andDismissBlock:(void(^ __nullable)(void))dismissBlock;
 
+/**
+ * Modally presents the feedback widget above the top visible view controller and executes given blocks.
+ * @discussion Calls to this method will be ignored if consent for @c CLYConsentFeedback is not given while @c requiresConsent flag is set on initial configuration.
+ * @param wigetCallback Block to be executed when widget is displayed/dismissed
```
Comment (AliRKat) [inline CountlyFeedbackWidget.h:48]: widgetCallback it should be i believe

File: CountlyFeedbackWidget.h [hunk 3]
```diff
@@ -41,6 +42,12 @@ extern NSString* const kCountlyReservedEventRating;
...
+ */
+- (void)presentWithCallback:(WidgetCallback) wigetCallback;
```
Comment (AliRKat) [inline CountlyFeedbackWidget.h:50]: widgetCallback it should be i believe

**Review by AliRKat — CHANGES_REQUESTED** [review]
there are 2 typos i spot other than those seemed fine with me. great work!

**Review by AliRKat — APPROVED** [review]

**Review by turtledreams — APPROVED** [review]

**Pull request merged**